x-pack/plugins/security_solution/server/endpoint/routes/metadata/query_builders.ts

```diff
@@ -68,14 +73,53 @@ async function getPagingProperties(
   };
 }
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-function buildQueryBody(request: KibanaRequest<any, any, any>): Record<string, any> {
+function buildQueryBody(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  request: KibanaRequest<any, any, any>,
+  unerolledHostIds: string[] | undefined
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+): Record<string, any> {
+  const filterUnenrolledHosts = unerolledHostIds && unerolledHostIds.length > 0;
   if (typeof request?.body?.filter === 'string') {
-    return esKuery.toElasticsearchQuery(esKuery.fromKueryExpression(request.body.filter));
+    const kqlQuery = esKuery.toElasticsearchQuery(esKuery.fromKueryExpression(request.body.filter));
+    return {
+      bool: {
+        must: filterUnenrolledHosts
+          ? [
+              {
+                bool: {
+                  must_not: {
+                    terms: {
+                      'host.id': unerolledHostIds,
```

Comment:
I think we can move this into the `filter` context. Using `filter` improves performance because ES doesn't have to calculate a score.

Reply:
must not is filter context

Reply:
Ah true. Thanks.